Mathlib/Analysis/Normed/Order/UpperLower.lean

```diff
@@ -20,15 +22,21 @@ set).
 
 We also prove lemmas specific to `ℝⁿ`. Those are helpful to prove that order-connected sets in `ℝⁿ`
 are measurable.
--/
 
+## TODO
+
+Is there a way to generalise `IsClosed.upperClosure_pi`/`IsClosed.lowerClosure_pi` so that they also
+apply to `ℝ`, `ℝ × ℝ`, `EuclideanSpace ι ℝ`? `_pi` has been appended to their names to disambiguate
+from the other possible lemmas, but we will want there to be a single set of lemmas for all
+situations.
```

Comment:
This comment shouldn't hold up this PR, but the relevant lemmas could just take a homeomorphism with a pi-type as an explicit argument. It's not a pretty solution, but I think it would work, and I believe we already have the necessary homeomorphisms available.

(On second thought, would it need to be an order isomorphism too? That probably rules out this idea as being useful)

Reply:
Yeah, it would also need to be an order isomorphism. That's the crux. We need a way to say "Order on this type is given by those coordinates", which sounds a lot like the `Inducing`/`Embedding` business in the topology part of the library.

Food for thought, I guess